### PR TITLE
rust-proto-new: deduplicate a number of server-related things

### PR DESCRIPTION
**Pull request by wimax-grapl**
I'd suggest *not* going through commit-by-commit as this includes a number of plugin-work-queue things that will be revived in a future PR. (This PR will just be a squash)

CHANGES:

1. Deduplicate the structs that look like
```
    struct PipelineIngressProto<T, E>
    where
        T: PipelineIngressApi<E>,
        E: ToString + 'static,
    {
        api_server: T,
        _e: PhantomData<E>,
    }
```
This is now called ApiDelegate. This name sucks.

2. Use crate's Status everywhere instead of tonic Status.
3. Introduce `rpc_translate_proto_to_native!` to cut down on repeated translation layer stuff.
4. Move plugin-registry to using init_health_service

### CODE REVIEW

File: src/rust/rust-proto-new/src/server_internals.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+pub trait Api {}
+
+/// This struct implements the internal gRPC representation of the server.
+/// We've implemented the service trait generated by tonic
+/// in such a way that it delegates to an externally supplied
+/// Api. This way all the protocol buffer compiler generated
+/// types are encapsulated, and the public API is implemented in terms of
+/// this crate's sanitized types.
+pub struct ApiDelegate<T, E>
```
Comment (jgrillo-grapl) [inline src/rust/rust-proto-new/src/server_internals.rs:11]: Maybe `GrpcApi<T, E>`?

**Review by jgrillo-grapl — APPROVED** [review]
Looks good to me overall, a few small nits here and there but seems structurally sound.

**Comment by wimax-grapl** [comment]
This will also accomplish https://github.com/grapl-security/issue-tracker/issues/910 - as of next few commits

File: src/rust/rust-proto-new/src/graplinc/grapl/api/pipeline_ingress/v1beta1.rs [hunk 2]
```diff
@@ -352,6 +316,7 @@ pub mod server {
     where
         T: PipelineIngressApi<E> + Send + Sync + 'static,
         E: ToString + Send + Sync + 'static,
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/rust-proto-new/src/graplinc/grapl/api/pipeline_ingress/v1beta1.rs:318]: E entirely removed in next commits

File: src/rust/rust-proto-new/src/server_internals.rs (hunk 1 above)
Comment (wimax-grapl) [reply src/rust/rust-proto-new/src/server_internals.rs:11]: done except <T>

File: src/rust/rust-proto-new/src/server_internals.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+where
+    E: Into<Status>,
+{
+    pub api_server: T,
+    _e: PhantomData<E>,
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/rust-proto-new/src/server_internals.rs:16]: done

File: src/rust/rust-proto-new/src/server_internals.rs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+impl From<crate::SerDeError> for tonic::Status {
+    fn from(e: crate::SerDeError) -> Self {
+        tonic::Status::unknown(e.to_string())
+    }
+}
+
+/// Deduplicates the translation layer code in our Server rpc functions.
+/// - turn a proto-request into a native-request
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/rust-proto-new/src/server_internals.rs:38]: done

File: src/rust/rust-proto-new/src/server_internals.rs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+impl From<crate::SerDeError> for tonic::Status {
+    fn from(e: crate::SerDeError) -> Self {
+        tonic::Status::unknown(e.to_string())
+    }
+}
+
+/// Deduplicates the translation layer code in our Server rpc functions.
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/rust-proto-new/src/server_internals.rs:37]: done, good point

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

**Pull request merged**